### PR TITLE
complete (except for get_domain(...)) 0.11.0 beacon chain spec update

### PR DESCRIPTION
**Pull request by tersec**
`get_domain(...)` in 0.11.0 [takes a BeaconState](https://github.com/ethereum/eth2.0-specs/blob/v0.11.0/specs/phase0/beacon-chain.md#get_domain) parameter, but https://github.com/status-im/nim-beacon-chain/blob/devel/beacon_chain/validator_pool.nim calls it in contexts without such a state parameter readily available. Potential refactoring, adjustment to get such a parameter would need to come after https://github.com/status-im/nim-beacon-chain/pull/819 which already refactors `validator_pool` (among other modules), so this PR will not fix `get_domain(...)`.

Finalizes locally.

It would be possible to fix it elsewhere, but then it would become potentially internally inconsistent, worse than either an internally consistent 0.10.1-like world or 0.11.0-like world.

### CODE REVIEW

**Comment by arnetheduck** [comment]
I started using non-`BeaconState` in some functions because the data needs to be passed across the beaconnode-validator API, so "minimal information" was the guideline (for domain, but really for signatures in general). 

I think it might make sense to continue this trend by starting to pass less that the full BeaconState and more specific data to a wider range of helpers which will make it easier to analyze the code for data dependencies etc - where it makes sense to do so - now that the suite is solid and the spec is nearly done.

**Comment by tersec** [comment]
I agree, that makes sense -- but with regard to this PR specifically, I'd still say the best ordering is this 0.11.0-without-`get_domain(...)` changes and https://github.com/status-im/nim-beacon-chain/pull/819 not having a mutual dependency, then when they both get in, resolving the remaining differences, rather than try to coordinate between these branches.

But, yes, for all sorts of reasons (including knowing when performance optimizations are safe, looking at the attack surface of any particular function or part of code, all the usual things) clarifying data flow is valuable, and now seems the time to do it, modulo other milestone priorities.

**Comment by tersec** [comment]
Well https://travis-ci.org/github/status-im/nim-beacon-chain/jobs/665444081 failed in some ephemeral/transient way I've not seen before, but that's a Travis issue.

File: beacon_chain/spec/helpers.nim [hunk 1]
```diff
@@ -114,13 +114,38 @@ func int_to_bytes4*(x: uint64): array[4, byte] =
...
+  # ``genesis_validators_root``.
+  # This is used primarily in signature domains to avoid collisions across
+  # forks/chains.
+  hash_tree_root(ForkData(
+    current_version: current_version,
+    genesis_validators_root: genesis_validators_root
+  ))
+
+# https://github.com/ethereum/eth2.0-specs/blob/v0.11.0/specs/phase0/beacon-chain.md#compute_fork_digest
+func compute_fork_digest(current_version: array[4, byte],
```
Comment (arnetheduck) [inline beacon_chain/spec/helpers.nim]: the array here is called `Version` in the spec - could potentially alias it..

Comment (tersec) [reply beacon_chain/spec/helpers.nim]: Sure, that's a bigger issue -- there are `TODO`s elsewhere noting that. I agree.

**Pull request merged**